src/AddIns/Uno.UI.MediaPlayer.Skia.Gtk/VideoView.cs

```diff
@@ -381,18 +383,24 @@ private void AssignWindowId()
 
 		private void DestroyChildWindow()
 		{
-			if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+			// DestroyChildWindow is called by Dispose (which is called by the destructor on a GC finalizer thread)
+			// and is also called by MediaPlayer property setter.
+			// The concurrent execution can cause memory corruption. So, we lock to avoid this issue.
+			lock (s_lock)
 			{
-				this.Log().Debug($"{GetHashCode():X8} Destroying child video window");
-			}
+				if (this.Log().IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+				{
+					this.Log().Debug($"{GetHashCode():X8} Destroying child video window");
+				}
 
-			RemoveWindowId();
+				RemoveWindowId();
 
-			if (_videoWindow is not null)
-			{
-				_videoWindow.Hide();
-				_videoWindow.Destroy();
-				_videoWindow = null;
+				if (_videoWindow is not null)
+				{
+					_videoWindow.Hide();
+					_videoWindow.Destroy();
+					_videoWindow = null;
+				}
```

Comment:
Good catch! As it is executed on a background thread, we may need to execute that on the main looper/dispatcher for GTK instead of serializing it.

It is likely that moving this code in something like this:

https://github.com/unoplatform/uno/blob/b831d0d39ffb0a7375876692c18ea1b678046f1c/src/Uno.UI.Runtime.Skia.Gtk/UI/GtkDispatch.cs#L9

Will be enough.

Reply:
Yes, that would make sense.

Reply:
Done!